tools/generate.js

```diff
@@ -0,0 +1,265 @@
+  export async function loadImplementations(isParallel = true) {
+    let modules;
+    IFSTATEMENT
+    return modules.then(([
+      THEN
+    ]) => {
+      IMPLEMENTEDINTERFACES
+      implementationLoaded = true;
+    });
+  }
+`);
+
+const buildExportFunction = (arg) => {
+  const baseTemplate = defaultTemplate(`
+    export function getInterfaceWrapper(name: '${arg.NAME}'): typeof ${arg.TYPE}; 
+  `);
+  return baseTemplate({}); 
+};
+
+const buildGetInterfaceWrapper = defaultTemplate(`
+  EXPORTFUNCTION
+  export function getInterfaceWrapper(name: string): any;
+  export function getInterfaceWrapper(name) {
+    if (!implementationLoaded) {
+      throw new Error('DOM Implementation not loaded');
+    }
+    return implementedInterfaces.get(name);
+  }
+`);
+
+const buildIntegration = defaultTemplate(`
+  HEADSTATEMENT
+  LOADIMPLEMENTATIONS
+  GETINTERFACEWRAPPER
+`);
+
+// Build the code
+const typeImports = moduleSpecifiers.map(specifier => buildTypeImports({
+  TYPE: specifier.type,
+  SPECIFIER: specifier.path,
+  ISDEFAULT: specifier.isDefault
+}));
+
+const headStatement = buildHeadStatement({
+  TYPEIMPORTS: typeImports
+});
+
+const parallelImports = moduleSpecifiers.map(specifier => buildParallelImports({
+  MODULE: specifier.path
+}));
+
+const sequentialImports = moduleSpecifiers.map(specifier => buildSequentialImports({
+  MODULE: specifier.path
+}));
+
+const parallelModule = buildModule({
+  SOURCE: t.arrayExpression(parallelImports.map(imp => imp.expression))
+});
+
+const sequentialModule = buildModule({
+  SOURCE: t.arrayExpression(sequentialImports.map(imp => imp.expression))
+});
+
+const ifStatement = buildIfStatement({
+  ISPARALLEL: t.identifier('isParallel'),
+  PARALLELMODULE: parallelModule,
+  SEQUENTIALMODULE: sequentialModule
+});
+
+// Template cannot handle single-word task,
+// so I choose to use string concatenation to solve this problem.
+const then = moduleSpecifiers.map(specifier => {
+  if (specifier.isDefault) {
+    return specifier.type;
+  } else {
+    return `{ ${specifier.type} }`;
+  }
+}).join(', ');
+
+const implementedInterfaces = moduleSpecifiers.map(specifier => buildImplementedInterfaces({
+  NAME: specifier.name,
+  ISDEFAULT: specifier.isDefault,
+  TYPE: specifier.type
+}));
+
+const loadImplementations = buildLoadImplementations({
+  IFSTATEMENT: ifStatement,
+  THEN: then,
+  IMPLEMENTEDINTERFACES: implementedInterfaces
+});
+
+const exportFunction = moduleSpecifiers.map(specifier => buildExportFunction({  
+  NAME: specifier.name,
+  TYPE: specifier.type
+}));
+
+const getInterfaceWrapper = buildGetInterfaceWrapper({
+  EXPORTFUNCTION: exportFunction
+});
+
+const integration = buildIntegration({
+  HEADSTATEMENT: headStatement,
+  LOADIMPLEMENTATIONS: loadImplementations,
+  GETINTERFACEWRAPPER: getInterfaceWrapper
+});
+
+// Generate the code
+const code = generate.default(t.program(integration)).code;
+const __dirname = 'src/living/';
```

Comment:
`__dirname` is an existed variable in a Node.js commonjs module, could you please change the name?